packages/relay-kit/src/packs/gelato/GelatoRelayPack.ts

```diff
@@ -131,11 +208,50 @@ export class GelatoRelayPack implements RelayPack {
     target,
     encodedTransaction,
     chainId,
-    options
+    options = {}
   }: RelayTransaction): Promise<RelayResponse> {
     const response = options.isSponsored
       ? this.sendSponsorTransaction(target, encodedTransaction, chainId)
       : this.sendSyncTransaction(target, encodedTransaction, chainId, options)
     return response
   }
+
+  /**
+   * Sends the Safe transaction to the Gelato Relayer for execution.
+   * If the Safe is not deployed, it creates a batch of transactions including the Safe deployment transaction.
+   *
+   * @param {SafeTransaction} safeTransaction - The Safe transaction to be executed.
+   * @param {Safe} safe - The Safe object related to the transaction.
+   * @returns {Promise<RelayResponse>} Returns a Promise that resolves with a RelayResponse object.
+   */
+  async executeRelayTransaction(
+    safeTransaction: SafeTransaction,
+    safe: Safe
+  ): Promise<RelayResponse> {
+    const isSafeDeployed = await safe.isSafeDeployed()
+    const chainId = await safe.getChainId()
+    const safeAddress = await safe.getAddress()
+    const safeTransactionEncodedData = await safe.getEncodedTransaction(safeTransaction)
+
+    if (isSafeDeployed) {
+      const relayTransaction: RelayTransaction = {
+        target: safeAddress,
+        encodedTransaction: safeTransactionEncodedData,
+        chainId
+      }
+
+      return this.relayTransaction(relayTransaction)
```

Comment:
`executeRelayTransaction` is not considering that the user could be executing an sponsored transaction or a transaction using syncTransaction. It only allows to use syncTransaction.

Reply:
done!